ludwig/datasets/loaders/av_mnist.py

```diff
@@ -0,0 +1,214 @@
+        """
+        Args:
+            file_paths (List[string]): The directory where all the .npy files are located for the training and test data.
+        Returns:
+            The array of directory paths
+        """
+        self.process_source_dataset(self.raw_dataset_dir)
+        for dataset in ["training", "testing"]:
+            if not self.modal_separate:
+                if dataset == "training":
+                    self.audio_data = np.load(os.path.join(self.raw_dataset_dir, "audio", "train_data.npy"))
+                    self.mnist_data = np.load(os.path.join(self.raw_dataset_dir, "image", "train_data.npy"))
+                    self.labels = np.load(os.path.join(self.raw_dataset_dir, "train_labels.npy"))
+                else:
+                    self.audio_data = np.load(os.path.join(self.raw_dataset_dir, "audio", "test_data.npy"))
+                    self.mnist_data = np.load(os.path.join(self.raw_dataset_dir, "image", "test_data.npy"))
+                    self.labels = np.load(os.path.join(self.raw_dataset_dir, "test_labels.npy"))
+
+                self.audio_data = self.audio_data[:, np.newaxis, :, :]
+                self.mnist_data = self.mnist_data.reshape(self.mnist_data.shape[0], 1, 28, 28)
+                self.write_output_dataset(self.labels, self.audio_data, os.path.join(self.raw_dataset_dir, dataset))
+                self.write_output_dataset(self.labels, self.mnist_data, os.path.join(self.raw_dataset_dir, dataset))
+            else:
+                if self.modal:
+                    if self.modal not in ["audio", "image"]:
+                        raise ValueError("the value of modal is allowed")
+                    if dataset == "train":
+                        self.data = np.load(os.path.join(self.raw_dataset_dir, self.modal, "train_data.npy"))
+                        self.labels = np.load(os.path.join(self.raw_dataset_dir, "train_labels.npy"))
+                    else:
+                        self.data = np.load(os.path.join(self.raw_dataset_dir, self.modal, "test_data.npy"))
+                        self.labels = np.load(os.path.join(self.raw_dataset_dir, "test_labels.npy"))
+
+                    if self.modal == "audio":
+                        self.data = self.data[:, np.newaxis, :, :]
+                    elif self.modal == "image":
+                        self.data = self.data.reshape(self.data.shape[0], 1, 28, 28)
+                    self.write_output_dataset(self.labels, self.data, os.path.join(self.raw_dataset_dir, dataset))
+                else:
+                    raise ValueError("the value of modal should be given")
+
+        return super().transform_files(file_paths)
+
+    def load_unprocessed_dataframe(self, file_paths: List[str]) -> pd.DataFrame:
+        """Load dataset files into a dataframe."""
+        return self.output_training_and_test_data()
+
+    def process_source_dataset(self, raw_dataset_dir="."):
```

Comment:
From my understanding, it seems like this doesn't take into account either `self.modal` or `self.modal_separate` while saving the files as npy files.  So, this will always create the following directory structure:

```
raw_dataset_dir
    |-- images
        |-- train_data.npy
        |-- test_data.npy
    |-- labels
        |-- train_labels.npy
        |-- test_labels.npy
```

This by itself is great, but will break if `self.modal == 'audio'` since the file to be loaded in `transform_files` won't exist

Reply:
Lets get the image workflow tested and working end to end first, then we'll focus on audio, just trying to get past smaller bits of functionality